source/MaterialXView/Viewer.cpp

```diff
@@ -1617,8 +1586,21 @@ void Viewer::drawContents()
         {
             mx::TextureBakerPtr baker = mx::TextureBaker::create();
             baker->setImageHandler(imageHandler);
-            baker->bakeShaderInputs(shaderRef, _genContext, _bakeFilename.getParentPath());
-            baker->writeBakedDocument(shaderRef, _bakeFilename);
+            mx::ShaderRefPtr shaderRef = material->getElement()->asA<mx::ShaderRef>();
+            if (shaderRef)
+            {
+                baker->bakeShaderInputs(shaderRef, _genContext, _bakeFilename.getParentPath());
+                baker->writeBakedDocument(shaderRef, _bakeFilename);
+            }
+            else
+            {
+                mx::NodePtr shader = material->getElement()->asA<mx::Node>();
```

Comment:
The difference between ILM's versiona and this one is that I'm copying over checking for material nodes vs shaderrefs for texture baking.